src/lib/list/selection-list.ts

```diff
@@ -274,6 +324,62 @@ export class MatSelectionList extends _MatSelectionListMixinBase
     }
   }
 
+  /** Reports a value change to the ControlValueAccessor */
+  _reportValueChange() {
+    if (this.options) {
+      this._onChange(this._getSelectedOptionValues());
+    }
+  }
+
+  /** Emits a change event if the selected state of an option changed. */
+  _emitChangeEvent(option: MatListOption) {
+    this.change.emit(new MatSelectionListChange(this, option));
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  writeValue(values: string[]): void {
+    if (this.options) {
+      this._setOptionsFromValues(values || []);
+    }
+  }
+
+  /** Implemented as a part of ControlValueAccessor. */
+  setDisabledState(isDisabled: boolean): void {
+    if (this.options) {
+      this.options.forEach(option => option.disabled = isDisabled);
+    }
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnChange(fn: any): void {
+    this._onChange = fn;
+  }
+
+  /** Implemented as part of ControlValueAccessor. */
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  /** Returns the option with the specified value. */
+  private _getOptionFromValue(value: string): MatListOption | undefined {
```

Comment:
Call this `_getOptionByValue`? Also it's used only in one place so it could be inlined.

Reply:
Sounds better. But I'd still like to keep it as a function. Otherwise it won't be that readable anymore.